### PR TITLE
chore: lock gh actions on sha instead of tag

### PR DESCRIPTION
**Pull request by npalm**
## Description

A good practice in supply chain security is to have immutable builds. Currently GH tags for actions are mutable, even when a specific tag like x.y.z is used. Therefore, this PR locks the actions on the SHA. Dependabot can update the actions based on SHA, including the comments poitning to the underlying version.

## Migrations required

NO

## Verification

n/a

### CODE REVIEW

File: .github/workflows/pr-opened.yml [hunk 1]
```diff
@@ -1,27 +1,17 @@
...
-            })
+          script: "// adds a comment to the PR (there is the issue API only which works work PRs too)\ngithub.rest.issues.createComment({\n  issue_number: context.issue.number,\n  owner: context.repo.owner,\n  repo: context.repo.repo,\n  body: 'Hey @${{ github.event.pull_request.user.login }}! \U0001F44B\\n\\nThank you for your contribution to the project. Please refer to the [contribution rules](blob/main/CONTRIBUTING.md) for a quick overview of the process.\\n\\nMake sure that this PR clearly explains:\\n\\n- the problem being solved\\n- the best way a reviewer and you can test your changes\\n\\nWith submitting this PR you confirm that you have the rights of the code added and agree that it will published under the [MIT license](https://github.com/npalm/terraform-aws-gitlab-runner/blob/main/LICENSE).\\n\\n_This message was generated automatically. You are welcome to [improve it](blob/main/.github/workflows/pr-opened.yml)._'\n})\n"
```
Comment (kayman-mk) [inline .github/workflows/pr-opened.yml:17]: issue: all lines have been combined into one starting with teh comment char `//`. Guess the script will no longer excecuted, right?

**Review by kayman-mk — CHANGES_REQUESTED** [review]

File: .github/workflows/pr-opened.yml (hunk 1 above)
Comment (npalm) [reply .github/workflows/pr-opened.yml:17]: Oops, will fix it. Used a script to convert the yml files

**Comment by npalm** [comment]
@kayman-mk can you check this tiny PR please?

**Review by kayman-mk — APPROVED** [review]
Guess the SHA256 values are correct ;-)

**Comment by npalm** [comment]
> Guess the SHA256 values are correct ;-)

Yup used a nice tool to generate them. And random check some

**Pull request merged**